src/Formidable.res

```diff
@@ -0,0 +1,565 @@
+open FormidableExtra
+
+module Events = FormidableEvents
+module Validations = FormidableValidations
+
+module type Type = {
+  type t
+}
+
+module type Values = {
+  include Type
+
+  let init: t
+}
+
+module type Handlers = {
+  type values
+
+  type error
+
+  let onSubmit: option<values => unit>
+
+  let onSubmitError: option<(values, list<error>) => unit>
+}
+
+module States = {
+  module Field = {
+    module Status = {
+      type t<'error> = [#pristine | #valid | #touched | #errors(list<'error>)]
+    }
+
+    @bs.deriving(accessors)
+    type t<'values, 'error> = {
+      status: Status.t<'error>,
+      validate: (option<Validations.Strategy.t>, 'values) => Status.t<'error>,
+    }
+
+    let isPristine = field =>
+      switch field {
+      | {status: #pristine} => true
+      | _ => false
+      }
+
+    let isValid = field =>
+      switch field {
+      | {status: #valid} => true
+      | _ => false
+      }
+
+    let isTouched = field =>
+      switch field {
+      | {status: #touched} => true
+      | _ => false
+      }
+
+    let hasErrors = field =>
+      switch field {
+      | {status: #errors(_)} => true
+      | _ => false
+      }
+
+    let getErrors = field =>
+      switch field {
+      | {status: #errors(errors)} => Some(errors)
+      | _ => None
+      }
+
+    let reset = field => {...field, status: #pristine}
+  }
+
+  let isPristine = fields => fields->Map.String.every(_ => Field.isPristine)
+
+  let isValid = fields => fields->Map.String.every(_ => Field.isValid)
+
+  let isTouched = fields => fields->Map.String.some(_ => Field.isPristine)
```

Comment:
`let isTouched = fields => fields->Map.String.some(_ => !Field.isPristine)`
Maybe?

Like "it's touched if _some_ field is _not_ pristine anymore?"

Reply:
this is true ! I will correct it thank you @gaku-sei -san

Reply:
oh maybe it should be 
```
  let isTouched = fields => fields->Map.String.some(_ => Field.isTouched)

```
right @gaku-sei -san

Reply:
Looks good 👍